### PR TITLE
Add difflint Makefile target

### PR DESCRIPTION
**Pull request by panpilkarz**
Introduce a `difflint` command which runs all python linters against modified/added files only.

___
The purpose of dummy `.difflint` file added to the repo is a need to provide at least a single file to all linters. Tried hard to have a condition in the Makefile to run linters only if any file is subjected for linting (check if `$(FILES)` variable is empty) but unsuccessful. Feel free to share a suggestion on a workaround. According to my research, this one should work but it doesn't:
```
difflint:
    $(eval FILES := $(shell git status --porcelain | grep --color=never -E $(DIFFLINT_RE) | awk '{print $$NF}'))
ifneq ($(FILES),)
    mypy $(FILES) .difflint
    black --check --diff $(FILES) .difflint
    flake8 $(FILES) .difflint
    isort $(FILES) .difflint --diff --check-only
    pylint $(FILES) .difflint
endif
```

### CODE REVIEW

File: .difflint [hunk 1]
```diff
@@ -0,0 +1 @@
+pass
```
Comment (fredo) [inline .difflint:1]: could we create and remove the `.difflint` file within the make target?

**Review by fredo — APPROVED** [review]
Just some nitpicks, otherwise looks good!

File: Makefile [hunk 2]
```diff
@@ -22,6 +23,14 @@ format: black
 	isort $(CODE_DIRS)
 	npx prettier --write $(CONTRACTS)
 
+difflint:
+	$(eval FILES := $(shell git status --porcelain | grep --color=never -E $(DIFFLINT_RE) | awk '{print $$NF}'))
```
Comment (istankovic) [inline Makefile:27]: Why not use `git ls-files -m`?

**Comment by istankovic** [comment]
> The purpose of dummy `.difflint` file added to the repo is a need to provide at least a single file to all linters. Tried hard to have a condition in the Makefile to run linters only if any file is subjected for linting (check if `$(FILES)` variable is empty) but unsuccessful. Feel free to share a suggestion on a workaround. According to my research, this one should work but it doesn't:
[...]

I have love/hate relationship with `make`. I would have fully expected your example to work.

Have you tried perhaps stripping the `FILES` variable to remove possible extra whitespace?

File: Makefile (hunk 2 above)
Comment (panpilkarz) [reply Makefile]: Took a look on that and would be nice to use it, however it shows only modified files, while added files are also needed in my opinion.

File: Makefile [hunk 3]
```diff
@@ -22,6 +23,14 @@ format: black
...
+	mypy $(FILES) .difflint
+	black --check --diff $(FILES) .difflint
+	flake8 $(FILES) .difflint
+	isort $(FILES) .difflint --diff --check-only
+	pylint $(FILES) .difflint
```
[the remark this answers is not in the document]
Comment (panpilkarz) [reply Makefile]: Added.

**Comment by istankovic** [comment]
> > Have you tried perhaps stripping the FILES variable
> 
> @istankovic yes I did. it really suks
> 
> Running all linters on that dummy file is just a second so not really a big deal.
> 
> ```
> $ time make pydifflint
> echo "pass" > ".difflint.py"
> mypy .difflint.py
> Success: no issues found in 1 source file
> black --check --diff .difflint.py
> All done! ✨ � ✨
> 1 file would be left unchanged.
> flake8 .difflint.py
> isort .difflint.py --diff --check-only
> pylint .difflint.py
> rm -rf ".difflint.py"
> 
> real    0m1.256s
> ```

That is all nice, but the dummy files should really not be needed. Here is a solution without them:
```
pydifflint:                                                                                                           
    $(eval PY_FILES := $(shell git status --porcelain | grep --color=never -E $(DIFFLINT_PY_RE) | awk '{print $$NF}'))
    @if [ "$(PY_FILES)" != "" ]; then \                                                                               
        mypy $(PY_FILES) && \                                                                                         
        black --check --diff $(PY_FILES) && \                                                                         
        flake8 $(PY_FILES) && \                                                                                       
        isort $(PY_FILES) --diff --check-only && \                                                                    
        pylint $(PY_FILES); \                                                                                         
    fi                                                                                                                
```

**Review by istankovic — CHANGES_REQUESTED** [review]
We should get rid of the dummy files, they are not really needed.

**Comment by istankovic** [comment]
@panpilkarz could you please squash all three commits into a single one?

**Review by istankovic — APPROVED** [review]
Looks great, thanks!

**Pull request merged**